src/main/java/com/todo/backend/Controller.java

```diff
@@ -1,54 +1,302 @@
 package com.todo.backend;
 
-import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
+import com.todo.backend.Metrics.LastMetrics;
+import com.todo.backend.ToDo.Priority;
+
+import java.text.ParseException;
+import java.time.Instant;
 import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
 import java.util.List;
+import java.util.Map;
 
+@CrossOrigin(origins = "*")
 @RestController
 public class Controller {
 
-    private List<String> items = new ArrayList<>();
+    // #region ################################ DATA BASE
+    private List<ToDo> DB = new ArrayList<>();
+    private Metrics metrics = new Metrics();
+
+    private SearchParams previousParams = new SearchParams();
+    private List<ToDo> lastResponse = null;
+    // #endregion
+
+    // #region ################################ FILTER & SORTING
+    private ResponseEntity<String> toggleDoneUndone(int id, Integer inPage, boolean markedAs) {
+        boolean found = false;
+
+        if (inPage == null) {
+            for (ToDo toDo : DB) {
+                if (toDo.getId() == id) {
+                    toDo.setDone(markedAs);
+                    found = true;
+                    break;
+                }
+            }
+        } else {
+            // IF inPage EXISTS, WE TOGGLE ALL FROM THE LAST RESPONSE (IF ANY)
+            List<ToDo> aux = this.lastResponse == null ? this.DB : this.lastResponse;
+
+            int maxpage = (int) Math.max(Math.ceil(aux.size() / 10f), 1);
+            int page = (int) Math.min(Math.max(inPage, 1), maxpage);
+            List<ToDo> slice = aux.subList(10 * (page - 1), Math.min(10 * page, aux.size()));
+
+            for (ToDo toDo : slice)
+                toDo.setDone(markedAs);
+
+            found = true;
+        }
+
+        this.metrics.needsRecalculate();
+        this.lastResponse = null; // FORCE RECALCULATE
+
+        if (found)
+            return new ResponseEntity<>("Success", HttpStatus.ACCEPTED);
+        return new ResponseEntity<>("Not Found", HttpStatus.NOT_FOUND);
+    }
+
+    private List<ToDo> filter(Boolean done, String text, Priority priority) {
+        List<ToDo> filtered = new ArrayList<>(); // CLEAR ALL
+
+        for (ToDo item : DB) {
```

Comment:
there are better ways to filter data in java, check `stream()`

https://www.geeksforgeeks.org/stream-filter-java-examples/